packages/go_router/lib/src/information_provider.dart

```diff
@@ -62,6 +63,10 @@ class RouteInformationState<T> {
   /// [NavigatingType.restore].
   final Completer<T?>? completer;
 
+  /// The completer that needs to be completed when route is replaced.
+  /// [completer] is ignored when replacing routes.
```

Comment:
maybe add only only used when type is NavigatingType.push or NavigatingType.pushReplacement
NavigatingType.replace